bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1128 @@
+                updateState(CHANNEL_2_ON_OFF_ONLY, resource.getOnOffState(), fullUpdate);
+                updateState(CHANNEL_2_ALERT, resource.getAlertState(), fullUpdate);
+                break;
+
+            case LIGHT_LEVEL:
+                updateState(CHANNEL_2_LIGHT_LEVEL, resource.getLightLevelState(), fullUpdate);
+                updateState(CHANNEL_2_LIGHT_LEVEL_ENABLED, resource.getEnabledState(), fullUpdate);
+                break;
+
+            case MOTION:
+                updateState(CHANNEL_2_MOTION, resource.getMotionState(), fullUpdate);
+                updateState(CHANNEL_2_MOTION_ENABLED, resource.getEnabledState(), fullUpdate);
+                break;
+
+            case RELATIVE_ROTARY:
+                if (fullUpdate) {
+                    addSupportedChannel(CHANNEL_2_ROTARY_STEPS);
+                }
+                updateState(CHANNEL_2_ROTARY_STEPS, resource.getRotaryStepsState(), fullUpdate);
+                break;
+
+            case TEMPERATURE:
+                updateState(CHANNEL_2_TEMPERATURE, resource.getTemperatureState(), fullUpdate);
+                updateState(CHANNEL_2_TEMPERATURE_ENABLED, resource.getEnabledState(), fullUpdate);
+                break;
+
+            case ZIGBEE_CONNECTIVITY:
+                updateConnectivityState(resource);
+                break;
+
+            case SCENE:
+                updateState(CHANNEL_2_SCENE, resource.getSceneState(), fullUpdate);
+                break;
+
+            default:
+                return false;
+        }
+        if (thisResource.getType() == ResourceType.DEVICE) {
+            updateState(CHANNEL_2_LAST_UPDATED, new DateTimeType(), fullUpdate);
+        }
+        return true;
+    }
+
+    /**
+     * Check the Zigbee connectivity and set the thing online status accordingly. If the thing is offline then set all
+     * its channel states to undefined, otherwise execute a refresh command to update channels to the latest current
+     * state.
+     *
+     * @param resource a Resource that potentially contains the Zigbee connectivity state.
+     */
+    private void updateConnectivityState(Resource resource) {
+        ZigbeeStatus zigbeeStatus = resource.getZigbeeStatus();
+        if (Objects.nonNull(zigbeeStatus)) {
+            logger.debug("{} -> updateConnectivityState() thingStatus:{}, zigbeeStatus:{}", resourceId,
+                    thing.getStatus(), zigbeeStatus);
+            hasConnectivityIssue = zigbeeStatus != ZigbeeStatus.CONNECTED;
+            if (hasConnectivityIssue) {
+                if (thing.getStatusInfo().getStatusDetail() != ThingStatusDetail.COMMUNICATION_ERROR) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
+                            "@text/offline.api2.comm-error.zigbee-connectivity-issue");
+                    supportedChannelIdSet.forEach(channelId -> updateState(channelId, UnDefType.UNDEF));
```

Comment:
@openhab/add-ons-maintainers - what is the best practice here? I have seen this logic before, but if setting all channel states to UNDEF when thing goes offline would be the only logical thing to do, the framework could do this. I understand the logic that we can't trust the state when losing connection, but on the other hand it could be just for a few seconds, which would be annoying when having a lot of linked items which would change states back and forth.

Reply:
I also do that in some of the bindings I am maintaining but not necessarily for all channels, it depends on channel context.

Reply:
@lolodomo - in this context, what would be your choice for:
- Battery-powered Hue motion sensor (channels like motion-detected, light-level, battery-level, etc.)
- Mains-powered Zigbee bulb (channels like brightness, color-temperature)

Keep state or UNDEF? And most importantly: The reasoning? 🙂

Reply:
Well my reasoning is that if you cannot communicate a thing, then its current state is not defined; this is a matter of pure philosophical logic. (But if you want to change that then I will do whatever you want..)

Reply:
It's not that I personally want anything in particular, but I would like to establish/understand some kind of logic/rule of thumb here.

Generally, if a thing is offline, we cannot trust the values of any of its channels. Perhaps unless you could have a write-only channel? Yet, core doesn't automatically set all channels to UNDEF, but leaves this choice to the individual bindings. I have seen different implementations/interpretations, and am now wondering pros and cons. Blindly setting everything to UNDEF have at least the con that flaky connections could lead to this "flipping" back and forth for all channels. At the same time it seems philosophically correct.

Other @openhab/add-ons-maintainers opinions/insights?

Reply:
@andrewfg - I believe the Hue binding didn't do this in the previous implementation?

Reply:
I actually have a rule for this behavior in my instance that's attached to most Things - when the Thing goes offline (for more than 5 seconds iirc -- to debounce quick reconnects), set all linked Items on that Thing to UNDEF. I'd love to see such an (optional) behavior implemented in core, and no bindings would have to deal with it. I don't realistically see that happening in 4.0 timeframe. I've not seen any other bindings that do it, but if there are already some that do, it would seem to currently be at the binding author's discretion, and shouldn't hold this PR. Easy to change in the future as a larger audience starts testing this, and if it becomes clear most users don't want this behavior, another PR could remove it quickly.

Reply:
👍. Please link it here, as I have additional thoughts from the perspective of the HomeKit addon.